packages/wdio-cli/src/index.js

```diff
@@ -5,25 +5,35 @@ import path from 'path'
 import yargs from 'yargs'
 
 import Launcher from './launcher'
-import { handler } from './commands/run'
+import { handler, builder } from './commands/run'
+
+const SUPPORTED_COMMANDS = ['config', 'install', 'repl', 'run']
+const DEFAULT_CONFIG_FILENAME = 'wdio.conf.js'
 
 export const run = async () => {
     const argv = yargs
         .commandDir('commands')
-        .demandCommand()
         .help()
 
+    /**
+     * parse CLI arguments according to what run expects
+     */
+    if (!process.argv.find((arg) => arg === '--help')) {
+        for (const [name, param] of Object.entries(builder)) {
+            argv.option(name, param)
+        }
+    }
+
     /**
      * if yargs doesn't run a command from the command directory
      * we verify if a WDIO config file exists in the given "path" parameter
      * if so, we assume the user ran `wdio path/to/wdio.conf.js` and we execute `wdio run` command
      */
     const params = { ...argv.argv }
-    const wdioConfPath = path.join(process.cwd(), params._[0])
-    const wdioConf = fs.existsSync(wdioConfPath)
-
-    if (wdioConf) {
-        return handler({ configPath: wdioConfPath, ...params })
+    if (!params._.find((param) => SUPPORTED_COMMANDS.includes(param))) {
+        params.configPath = path.join(process.cwd(), params._[0] || DEFAULT_CONFIG_FILENAME)
```

Comment:
@christian-bromann this `path.join` appears to break support for absolute config paths, are they no longer supported?

Reply:
@diachedelic please raise an issue 1

Reply:
See #4585

Reply:
I am working on changing this file in [this](https://github.com/webdriverio/webdriverio/pull/4570) pull request. I'll take a look at #4585 as well.